### ISSUE TITLE
[WIP] Resolve accessibility test failures in SPA project templates

### ISSUE THREAD

**Issue by mkArtakMSFT** [dotnet/aspnetcore#25352]
**This issue will split into multiple - per each template area.**

CTI has reported a set of accessibility failures related to our default templates.:

## Angular Project Template failures
- [ ] [Bug 1162883: A11y_.NETCoreJuly2020_ASP.Net Core_Manageyouraccountprofile_AI4W: Heading levels assigned to the text content not in proper sequence.](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1162883)
    - [@SteveSandersonMS] This is the identity UI. I have looked into it, but there are *many* places it would need to be changed. A change as disruptive as this would be quite risky this close to release, and should involve the identity owners in its design. I've filed https://github.com/dotnet/aspnetcore/issues/25674. Since this doesn't stop the narrator from working correctly, I would recommend this be addressed after 5.0 ships, however @mkArtakMSFT may have views on whether that will be acceptable.
- [x] [Bug 1162890: A11y_.NETCoreJuly2020_ASP.Net Core_Manageyouraccountprofile_Keyboard: Logout button receives keyboard focus but focus is not visible when navigating to it using keyboard.](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1162890)
    - [@SteveSandersonMS] I've diagnosed this at https://github.com/dotnet/aspnetcore/issues/25675. I'm providing a suggested fix for 5.0. Longer term, the identity owners might wish to choose a different HTML element structure to avoid the underlying problem and eliminate the need for a custom CSS override.
- [ ] [Bug 1163086: A11y_.NETCoreJuly2020_ASP.Net Core_Angular Counter_ScreenReader: Page title is not appropriate according to the pages that are loaded on the screen.](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1163086)
    - [@SteveSandersonMS] Doing this requires a new feature in the templates. I've filed https://github.com/dotnet/aspnetcore/issues/25676 to track it.
- [ ] [Bug 1163097: A11y_.NETCoreJuly2020_ASP.Net Core_Angular Fetch Data_ScreenReader: No association between the cells and column header present in the table.](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1163097)
    - [@SteveSandersonMS] This seems like a limitation in the narrator, not in the application. I've posted further details at https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1163097 and recommended that it be closed as external.
- [x] [Bug 1163139: A11y_.NETCoreJuly2020_ASP.Net Core_Angular Login_ScreenReader: How to access the eye symbol in the password field is not being announced by screen reader.](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1163139)
   - [@SteveSandersonMS] Closed as external.
- [x] [Bug 1163642: A11y_.NETCoreJuly2020_ASP.Net Core_Angular Login_ScreenReader: The "this article" link isn't descriptive enough while announced by screen reader.](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1163642)

## React Project Template Failures
- [x] [Bug 1162746: A11y_.NETCoreJuly2020_ASP.Net Core_React Register_ScreenReader: Error message displayed on the screen is not being announced by the screen reader.](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1162746)
   - [@SteveSandersonMS] Closed as no-repro.
- [x] [Bug 1163969: A11y_.NETCoreJuly2020_ASP.Net Core_React Register_ScreenReader: Required fields "email id" and "password" are not being announced as required by screen reader.](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1163969)
   - [@SteveSandersonMS] Closed as duplicate of https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1162746

## WebApp Project Template Failures
- [x] [Bug 1163117: A11y_.NETCoreJuly2020_ASP.Net Core_WebAppMVC Home_Keyboard: Focus order for the header items on the page is improper.](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1163117)

## Blazor Component Failures
- [x] [Bug 1163184: A11y_.NETCoreJuly2020_ASP.Net Core_Home_Zoom: At 200% page zoom the "About" link is not visible on the screen.](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1163184)
- [ ] [Bug 1163186: A11y_.NETCoreJuly2020_ASP.Net Core_Home_AI4D: The control type of the "Home" menu item is incorrect](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1163186)
   - [@SteveSandersonMS] This one does not seem correct to me. I've posted an additional comment at https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1163186 explaining why and suggesting that issue be closed.
- [ ] [Bug 1163198: A11y_.NETCoreJuly2020_ASP.Net Core_InputDateSample_ScreenReader: The narrator does not announce the selected date after the users selects the date from the "Select date" control](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1163198)
   - [@SteveSandersonMS] This one does not seem correct to me. I've posted an additional comment at https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1163198 explaining why and suggesting that issue be closed.

**Comment by SteveSandersonMS** [comment]
@mkArtakMSFT The ones I've checked off in the list above are either now closed or will be resolved once https://github.com/dotnet/aspnetcore/pull/25678 is merged. The remaining items are things that either:

 * I believe should be closed as invalid and have written up details on why
 * Or, in the case of the first one in the list, I don't think should be resolved in 5.0 and have filed a tracking issue for

**Comment by mkArtakMSFT** [comment]
Thanks for handling these, @SteveSandersonMS. I've resolved all the mentioned AzDO issues  per your recommendation. The only one we will need to address besides those you have in PR is #25676. Given the SLAs we have around accessibility issues we will need to address this even in a patch, if necessary.
